src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -145,6 +146,12 @@ public PerLeafResult searchLeaf(LeafReaderContext context, int k) throws IOExcep
             Map<Integer, Float> result = doExactSearch(context, new BitSetIterator(filterBitSet, cardinality), cardinality, k);
             return new PerLeafResult(filterWeight == null ? null : filterBitSet, result);
         }
+        /*
+         * If filters match all docs in this segment, then there is no need to do any extra step
+         * and should directly do ANN Search*/
+        if (filterWeight != null && cardinality == maxDoc) {
+            return new PerLeafResult(new FixedBitSet(0), doANNSearch(context, new FixedBitSet(0), 0, k));
```

Comment:
I see lots of `new FixedBitSet(0)` in KNNWeight. 
Could you factor it out with a meaningful name as a static variable then pass it down to the downstream?

```
private static final FixedBitSet ... = new FixedBitSet(0);
```